app/javascript/components/cloud-volume-form/cloud-volume-form.schema.js

```diff
@@ -0,0 +1,53 @@
+import { componentTypes, validatorTypes } from '@@ddf';
+import { parseCondition } from '@data-driven-forms/react-form-renderer';
+
+const createSchema = (fields, edit, loadSchema) => {
+  const idx = fields.findIndex((field) => field.name === 'volume_type');
+  const supports = edit ? 'supports_cloud_volume' : 'supports_cloud_volume_create';
+
+  return ({
+    fields: [
+      {
+        component: componentTypes.SELECT,
+        name: 'ems_id',
+        id: 'ems_id',
+        label: __('Storage Manager'),
+        placeholder: __('<Choose>'),
+        onChange: (value) => API.options(`/api/cloud_volumes?ems_id=${value}`).then(loadSchema()),
+        loadOptions: () => API.get(`/api/providers?expand=resources&attributes=id,name,${supports}&filter[]=${supports}=true`).then(({ resources }) =>
+          resources.map(({ id, name }) => ({ value: id, label: name }))),
+        isDisabled: edit,
```

Comment:
Make the field disabled if the storage manager is implicitly selected by the navigation source's context:

```suggestion
        isDisabled: edit || ems,
```